### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by manojbaishya** [spyder-ide/spyder#13565]
### What steps will reproduce the problem?

1. Install via Anaconda3-2020.07-Linux-x86_64.sh in ~/.local/apps/anaconda
2. Start spyder from terminal

Error keeps appearing after two fresh reinstalls.

### Traceback
```python-traceback
  File "/home/manoj/.local/apps/anaconda/lib/python3.8/site-packages/spyder/plugins/editor/plugin.py", line 1599, in update_todo_menu
    for text, line0 in results:
TypeError: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/home/manoj/.local/apps/anaconda/lib/python3.8/site-packages/spyder/plugins/editor/plugin.py", line 1582, in update_warning_menu
    check_results = editor.get_current_warnings()
AttributeError: 'NoneType' object has no attribute 'get_current_warnings'
```

## Versions

* Spyder version: 4.1.4 
* Python version: 3.8.3
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Linux 5.4.0-42-generic

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.5.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.16.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
keyring                        :  None (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.5.3 (OK)
pyls >=0.34.0;<1.0.0           :  0.34.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.5 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.1.2 (OK)
spyder_kernels >=1.9.2;<1.10.0 :  1.9.2 (OK)
watchdog                       :  None (OK)
xdg >=0.26                     :  0.26 (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.2.2 (OK)
numpy >=1.7                    :  1.18.5 (OK)
pandas >=0.13.1                :  1.0.5 (OK)
scipy >=0.17.0                 :  1.5.0 (OK)
sympy >=0.7.3                  :  1.6.1 (OK)
```

**Comment by manojbaishya** [comment]
Issue persists after creating a new environment and fresh install:
```
conda create -n test-env
conda activate test-env
conda install spyder=4.1.4
```

I am on Linux Mint 20.

**Comment by ccordoba12** [comment]
Please close Spyder, run in a terminal

    spyder --reset

and try again.

**Comment by manojbaishya** [comment]
Thank you, that fixed it! Apologies for the laziness, I see it is already mentioned in point 5 in the troubleshooting guide/FAQ.

I was able to retrigger the error and found out that the root cause was setting up the startup directory in the "current working directory" settings to a _symlinked folder_ from ~/ residing on another partition mounted at /media/user/. I also used another symlink to a folder residing on the same partition as the OS to verify if the separate drive was causing it but the error occurred again, so I am guessing the symlink itself caused the issue. Anyway, closing this.
